src/Centreon/Domain/HostConfiguration/HostConfigurationService.php

```diff
@@ -504,4 +489,104 @@ private function createHostGroupsBeforeLinking(array $groups): void
             }
         }
     }
+
+    /**
+     * The host name is checked to remove any illegal characters that monitoring server cannot accept.
+     *
+     * @param Host $host
+     * @throws HostConfigurationException
+     * @throws \Centreon\Domain\Engine\EngineException
+     */
+    private function checkIllegalCharactersInHostName(Host $host): void
+    {
+        $this->info('Check illegal characters in host name');
+        $engineConfiguration = $this->engineConfigurationService->findEngineConfigurationByName(
+            $host->getMonitoringServer()->getName()
+        );
+        if ($engineConfiguration === null) {
+            throw new HostConfigurationException(_('Unable to find the Engine configuration'));
```

Comment:
Shouldn't you create a static method in HostConfigurationServiceException for this one ?